src/playlist/cue/CueParser.cxx

```diff
@@ -302,10 +301,7 @@ CueParser::Get() noexcept
 		assert(current == nullptr);
 
 		finished = std::move(previous);
-		previous.reset();
 	}
 
-	auto result = std::move(finished);
-	finished.reset();
-	return result;
+	return std::move(finished);
```

Comment:
Yeah. I tested with godbolt to make sure. my understanding of std::move() of a unique_ptr is well defined.

Reply:
"Tested with godbolt" is no guarantee. That's proof by ancedote. Any future compiler/library version may have a different behavior.

Reply:
Sure. It gets optimized to the same code. Different compiler options can have an impact on size/speed but not behavior.

Reply:
True, but what gives this guarantee?

Reply:
I remember seeing a section in the C++ standard. I’ll dig it up when I can.

Reply:
There may be, I don't know if the standard guarantees this. But we should be sure before writing such code, and not just rely on anecdotal "proof".
Because it would be possible to implement the move operator by just swapping the two operands (which I do when I implement move operators, because that's the cheapest possible implementation). Which would break your code.

Reply:
Section 23.11.1 of the C++17 standard says:

```
Additionally, u can, upon request, transfer ownership to another unique pointer u2. Upon completion of such
a transfer, the following postconditions hold:
—(4.1) u2.p is equal to the pre-transfer u.p,
—(4.2) u.p is equal to nullptr, and
—(4.3) if the pre-transfer u.d maintained state, such state has been transferred to u2.d.
```

There's similar wording in the C++11 one.

Reply:
Agree, then this is safe. Thanks for the investigation, I learned something, too.